specifyweb/context/views.py

```diff
@@ -570,14 +570,34 @@ def views(request):
 @login_maybe_required
 @cache_control(max_age=86400, private=True)
 def viewsets(request):
-    """Retrive a list of Specify 6 viewset xml files."""
-    viewsets = []
-    for root, dir, files in os.walk(settings.SPECIFY_CONFIG_DIR):
-        for file in files:
-            if file.endswith('.views.xml'):
-                viewsets.append(
-                    os.path.relpath(os.path.join(root, file),settings.SPECIFY_CONFIG_DIR)
-                )
+    """
+    Retrieve a list of Specify 6 viewset xml files.
+    Filter out viewsets that are not applicable to the user type.
```

Comment:
Rather than filter out forms not applicable to current user type, can you filter out forms that contain a user type that is higher in power than the user type of current user?

this is because we would still want manager to be able to see and edit guest views

or to keep it simple, just include view if the path includes any user type (see https://github.com/specify/specify7/issues/4178#issuecomment-1788308291)

Reply:
Managers can _still_ see guest form definitions in this PR, which points to this not functioning as the comment indicates. That's a good thing!

Reply:
Sorry, I'll rephrase the comment to be more clear.  A Manager sees Manager and Guests files, while a Guest just sees Guest files, along with files without a user type.